packages/application-shell/src/components/user-settings-menu/user-settings-menu.js

```diff
@@ -62,136 +67,193 @@ function stateReducer(state, changes) {
   }
 }
 
-export class UserSettingsMenu extends React.PureComponent {
-  static displayName = 'UserSettingsMenu';
+const OptionalFeatureToggle = props =>
+  props.featureToggle ? (
+    <ToggleFeature flag={props.featureToggle}>{props.children}</ToggleFeature>
+  ) : (
+    props.children
+  );
+OptionalFeatureToggle.displayName = 'OptionalFeatureToggle';
+OptionalFeatureToggle.propTypes = {
+  featureToggle: PropTypes.string,
+  children: PropTypes.element.isRequired,
+};
+
+const renderLabel = (menu, locale) => {
+  const localizedLabel = menu.labelAllLocales.find(
+    loc => loc.locale === locale
+  );
+  if (localizedLabel) return localizedLabel.value;
+  return NO_VALUE_FALLBACK;
+};
 
-  static propTypes = {
-    firstName: PropTypes.string,
-    lastName: PropTypes.string,
-    email: PropTypes.string,
-    gravatarHash: PropTypes.string.isRequired,
-  };
+const UserSettingsMenuBody = props => {
```

Comment:
This component is basically rendered only when the user opens the menu, therefore we can connect to fetch the menu config on this component, hence the split of components